libdrgn/python/helpers.c

```diff
@@ -249,3 +250,267 @@ PyObject *drgnpy_linux_helper_pgtable_l5_enabled(PyObject *self, PyObject *args,
 		return PyErr_Format(PyExc_ValueError, "not Linux kernel");
 	Py_RETURN_BOOL(prog->prog.vmcoreinfo.pgtable_l5_enabled);
 }
+
+static void GenericIterator_dealloc(GenericIterator *self)
+{
+	if (self->iterator) {
+		self->iterator_deinit(self->iterator);
+		free(self->iterator);
+	}
+	Py_XDECREF(self->prog);
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static PyObject *GenericIterator_next(GenericIterator *self)
+{
+	return self->next(self);
+}
+
+PyTypeObject GenericIterator_type = {
+	PyVarObject_HEAD_INIT(NULL, 0)
+	.tp_name = "_drgn._GenericIterator",
+	.tp_basicsize = sizeof(GenericIterator),
+	.tp_dealloc = (destructor)GenericIterator_dealloc,
+	.tp_flags = Py_TPFLAGS_DEFAULT,
+	.tp_iter = PyObject_SelfIter,
+	.tp_iternext = (iternextfunc)GenericIterator_next,
+};
+
+static PyObject *for_each_task_next(GenericIterator *self)
+{
+	struct drgn_error *err;
+	struct task_iter *iterator = (struct task_iter *)self->iterator;
```

Comment:
This cast is unnecessary (same thing in the other `next` functions).
```suggestion
	struct task_iter *iterator = self->iterator;
```